test/end-to-end/offer.js

```diff
@@ -1103,4 +1109,400 @@ describe("Offer endpoint tests", () => {
+                    const FieldValidatorTester = BodyValidatorTester("location");
+                    FieldValidatorTester.mustBeString();
+                });
+            });
+        });
+
+        describe("testing as admin without god token", () => {
+            beforeAll(async () => {
+                await test_agent.post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+            });
+
+            test("should edit as an admin", async () => {
+                await test_agent.post(`/offers/edit/${future_test_offer._id}`)
+                    .expect(HTTPStatus.OK);
+            });
+        });
+
+
+        describe("testing as a company", () => {
+
+            describe("testing as another company", () => {
+
+                const test_user_company_2 = {
+                    email: "company2@email.com",
+                    password: "password123",
+                };
+
+                let test_company_2;
+
+                beforeAll(async () => {
+                    test_company_2 = await Company.create({
+                        name: "test company",
+                        bio: "a bio",
+                        contacts: ["a contact"]
+                    });
+                    await Account.create({
+                        email: test_user_company_2.email,
+                        password: await hash(test_user_company.password),
+                        company: test_company_2._id
+                    });
+
+                    await test_agent.post("/auth/login")
+                        .send(test_user_company_2)
+                        .expect(HTTPStatus.OK);
+                });
+
+                test("should fail if the company is not the owner", async () => {
+                    const res = await test_agent
+                        .post(`/offers/edit/${future_test_offer._id}`)
+                        .expect(HTTPStatus.FORBIDDEN);
+                    expect(res.body).toHaveProperty("reason", ValidationReasons.NOT_OFFER_OWNER(future_test_offer._id));
+                });
+
+            });
+
+            describe("testing as offer's owner", () => {
+                beforeAll(async () => {
+                    await test_agent.post("/auth/login")
+                        .send(test_user_company)
+                        .expect(HTTPStatus.OK);
+
+                });
+
+                test("should edit title", async () => {
+                    const edits = { "title": "This is a new title" };
+                    const res = await test_agent.post(`/offers/edit/${clean_test_offer._id}`)
+                        .send(edits)
+                        .expect(HTTPStatus.OK);
+                    expect(res.body).toMatchObject({
+                        ...clean_test_offer,
+                        ...edits
+                    });
+                    clean_test_offer = res.body;
+                    delete clean_test_offer["updatedAt"];
```

Comment:
this test should not mutate an offer that is available to other tests, as it can cause bugs in the future for someone who doesn't know how the tests are done. It would be really hard to debug

Reply:
Alternatively, you should create an offer within this describe and edit that instead, or, if the clean_test_offer is not used anywhere else, move it to this describe block